File: python/problem_51.py
import random
from typing import List


def shuffle(deck: List[int]) -> None:
    for i in range(len(deck) - 1, 0, -1):
        j = random.randint(0, i)
        deck[i], deck[j] = deck[j], deck[i]


def main():
    print("Problem 51:")
    deck = list(range(52))
    shuffle(deck)
    assert sorted(deck) == list(range(52))
    print("All Tests Passed")

File: typescript/problem_51.ts
export function shuffle(deck: number[]): void {
    for (let i = deck.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [deck[i], deck[j]] = [deck[j], deck[i]];
    }
}

export function problem51(): void {
    console.log("Problem 51:");
    const deck = Array.from({length: 52}, (_, i) => i);
    shuffle(deck);
    if (JSON.stringify([...deck].sort((a,b)=>a-b)) !== JSON.stringify(Array.from({length:52},(_,i)=>i)))
        throw new Error("assertion failed");
    console.log("All Tests Passed");
}
